bench/bench-perf.hs

```diff
@@ -0,0 +1,304 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE QuasiQuotes #-}
+
+module Main where
+
+import Control.Monad.State.Strict (liftIO)
+import Data.ByteString.UTF8 as BSU 
+import Data.Either
+import Data.Function
+import Data.Maybe
+import Data.String.Here
+import EVM
+import EVM.ABI
+import EVM.Effects (App, runApp)
+import EVM.Fetch qualified as Fetch
+import EVM.Solidity
+import EVM.Stepper qualified as Stepper
+import EVM.Transaction qualified
+import EVM.Types
+import EVM.UnitTest
+import Test.Tasty.Bench
+import Control.Monad.ST
+import EVM.FeeSchedule (feeSchedule)
+
+-- benchmark hevm using tasty-bench
+
+vmFromRawByteString :: App m => ByteString -> m (VM Concrete RealWorld)
+vmFromRawByteString bs = liftIO $
+  bs
+    & ConcreteRuntimeCode
+    & RuntimeCode
+    & initialContract
+    & vm0
+    & stToIO
+    & fmap EVM.Transaction.initTx
+
+vm0 :: Contract -> ST s (VM Concrete s)
+vm0 c = makeVm $ vm0Opts c
+
+vm0Opts :: Contract -> VMOpts Concrete
+vm0Opts c =
+  VMOpts
+    { contract = c,
+      calldata = (ConcreteBuf "", []),
+      value = Lit 0xfffffffffffff, -- balance
+      baseState = EmptyBase,
+      address = LitAddr 0xacab,
+      caller = LitAddr 0,
+      origin = LitAddr 0,
+      gas = 0xffffffffffffffff,
+      baseFee = 0,
+      priorityFee = 0,
+      gaslimit = 0xffffffffffffffff,
+      coinbase = LitAddr 0,
+      number = Lit 0,
+      timestamp = Lit 0,
+      blockGaslimit = 0xffffffffffffffff,
+      gasprice = 0,
+      maxCodeSize = 0xffffffff,
+      prevRandao = 0,
+      schedule = feeSchedule,
+      chainId = 1,
+      create = False,
+      txAccessList = mempty, -- TODO: support me soon
+      allowFFI = False,
+      otherContracts = [],
+      freshAddresses = 0,
+      beaconRoot = 0
+    }
+
+-- why do we need this?
```

Comment:
Do we have an answer to this question?

Reply:
Well, we need to figure out if we need it :)

If yes, then drop the comment. If no, then drop the whole function :)

Reply:
I guess we need it, so you can just drop the comment :)